frog/imports/ui/graph/Graph.jsx

```diff
@@ -0,0 +1,745 @@
+          Operator
+          <pre>{
+            JSON.stringify({'operatorType': operator.operatorType, 'type': operator.type, 'data': operator.data}, null, 2)
+          }</pre>
+        </ReactTooltip>
+      })}
+      {activities.map( (activity, i) => {
+        return <ReactTooltip
+          key={'actip' + i}
+          id={'tip'+activity._id}
+          place='bottom'
+          type='light'>
+          Activity: {activity._id}
+          <pre>{
+            //dividing by horizontalZoom since we only want the conversion seconds -> unit
+            JSON.stringify({'Data': activity.data, 'Activity Type': activity.activityType, 'Beginning': convertTimeToPx(scale, activity.position.x)/horizontalZoom + ' ' + scale}, null, 2)
+          }</pre>
+        </ReactTooltip>
+      })}
+    </span>
+  )
+}
+
+
+const DragAc = ( {activity, position, plane}) => {
+  return (
+    <Draggable
+      position= {position}
+      axis='both'
+      disabled= {false}>
+        <div data-tip data-for='dragac_tip' data-event-off='mouseDown'
+            style={divStyleNeg(activity)}>
+          {activity.data.name}
+          <ReactTooltip
+            id='dragac_tip'
+            type='light'
+            effect='solid'>
+            <pre>{JSON.stringify({'Data': activity.data, 'Activity Type': activity.activityType}, null, 2)}</pre>
+          </ReactTooltip>
+        </div>
+    </Draggable>
+  )
+}
+
+
+const BoxAc = ( {onHoverStart, hoverStop, plane, activity} ) => {
+
+  return(
+    <div
+      id={'box' + activity._id}
+      style={divStyleAc()}
+      onMouseOver={onHoverStart}
+      onMouseUp={hoverStop}>
+      {activity.data.name}
+    </div>
+  )
+}
+
+const RenderDraggable = ( { handleHover, handleHoverStop, activities}) => {return(
+    <div>
+      <div style={divListStyle}>
+
+        {activities.map((activity, i) => {
+          return <BoxAc
+          onHoverStart={(event) => handleHover(event, i%3 +1, activity)}
+          hoverStop={handleHoverStop}
+          key={i}
+          activity={activity}
+          plane={i%3 +1} /> //TODO to be changed with the real plane
+        })}
+      </div>
+
+    </div>
+
+  )
+}
+
+const TempAc = ({handleDragStop, position, plane, current}) => {
+  return (
+    <div id='dragac' style={{ position: 'absolute', zIndex: 2}} onMouseUp={(event) => handleDragStop(event, plane, current)}>
+      {current ?
+      <div  style={{ position: 'absolute'}}>
+        <DragAc
+          activity={current}
+          plane={plane}
+          position={position}
+        />
+      </div>
+    : '' }
+    </div>
+  )
+}
+
+
+export const RenderGraph = ( {
+  editorMode,
+  activities,
+  sizes,
+  operators,
+  onClickOperator,
+  clickedOperator,
+  listAvailableOperators,
+  deleteAc,
+  handleStop,
+  sourceOperator,
+  targetOperator,
+  loaded,
+  plane,
+  graphId,
+  moveCursor,
+  cursor,
+  scale,
+  activitySourceClicked}) => {
```

Comment:
That's a lot of props ;)  I understand that there are many thing going on. In that case, be sure to not use props when other solutions would be more efficient/readable. For example you can use something like Redux (http://redux.js.org/, I think Stian mentioned it) to save global states to be used by your react components. That way, you don't have to pass 20 props through 5 layers of nested components.